test/intercept/XMLHttpRequest/XMLHttpRequest.test.ts

```diff
@@ -218,3 +218,19 @@ test('intercepts an HTTP PATCH request', async () => {
   expect(request?.url.searchParams.get('userId')).toEqual('123')
   expect(request?.headers).toHaveProperty('x-custom-header', 'yes')
 })
+
+test('responds with timeout error', async () => {
+  const onTimeout = jest.fn()
+  try {
+    await xhr('GET', 'http://httpbin.org/get?userId=123', {
+      headers: {
+        'x-custom-header': 'yes',
+      },
+      timeout: 1,
+      onTimeout,
+      async: true,
+    })
+  } catch (error) {
```

Comment:
By extending our `xhr` util, we also make our test behave not like an actual XHR instance would behave. XHR won't throw on timeout, and neither should our test.